File: backend/src/main/java/com/ufg/stage/gate/tcc/models/entities/Meeting.java
package com.ufg.stage.gate.tcc.models.entities;

import com.ufg.stage.gate.tcc.models.enums.MeetingStatusEnum;
import com.ufg.stage.gate.tcc.models.enums.MeetingTypeEnum;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.Set;
import java.util.UUID;

@Entity
@Getter
@Setter
public class Meeting {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Enumerated(EnumType.STRING)
    private MeetingTypeEnum type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "professor_id")
    private User professor;

    private short stageGateNumber;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "meeting_participant",
            joinColumns = @JoinColumn(name = "meeting_id"),
            inverseJoinColumns = @JoinColumn(name = "participant_user_id")
    )
    private Set<User> participants;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id")
    private Project project;

    private LocalDate scheduleDate;

    private LocalTime startTime;

    private LocalTime endTime;

    @Enumerated(EnumType.STRING)
    private MeetingStatusEnum status;

    @OneToOne(mappedBy = "meeting", fetch = FetchType.LAZY)
    private MeetingReport report;

    @Column(name = "is_report_reminder_sent", columnDefinition = "boolean default false")
    private boolean isReportReminderSent;
}

File: front/src/app/pages/meetings/entities/meeting.ts
import { MeetingTypeEnum, MeetingStatusEnum } from '@/pages/meetings/entities/meetingListing';

export interface Meeting {
    id?: string;
    projectId: string;
    professorId: string;
    scheduleDate: Date | string;
    startTime: string;
    endTime: string;
    type: MeetingTypeEnum;
    status: MeetingStatusEnum;
    stageGateNumber: number;
    participants?: string[];
}

export interface CreateMeetingRequest {
    scheduleDate: string; // formato YYYY-MM-DD
    timeSlotId: string;
    type: MeetingTypeEnum;
}
